Preprod - v1.3.1

master-of-none opened the pull request:
# October Release – UI Changes

##  Summary
This PR introduces the **October 2025 UI updates** for Audemy.  
- Improved layout for accessibility (targeting blind users).  
- Fixed and improved many existing games and added new Games. 
- Enhanced navigation flow in the GameZone and Dashboard.  
- Fixed spacing and responsiveness issues across devices.  

---

## Testing Steps
1. Pull this branch and run locally.  
2. Navigate through all pages and check for:  
   - Correct alignment and spacing  
   - Proper rendering on desktop + mobile viewports  
   - GameZone accessibility  

---

## Release Notes
- October UI overhaul completed.  
- Production-ready after QA approval.  
- Major fixes to captions. 
- Various bugs squashed wrt UI/UX. 
- Dynamic rendering of images and added props to game pages. 
- Game codes and questions bank have been re written with decomposables logic and new questions have been added. 
- New Games have been added and updated. 

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds on-screen captions with progress and answer feedback across games, overhauls the Troubleshooting page with responsive steps, updates game assets/icons and styles, and refreshes content (Car Counting logic, Odd One Out DB, volunteers/press).
> 
> - **Games UX/Accessibility**:
>   - **Captions & Feedback**: `GameHeader.vue` redesigned to show question captions, progress bar, and correctness feedback; integrated across all game pages via new props (`showCaptions`, `isAnswerPlaying`, `isCorrect`, `firstMatchingAnswer`).
>   - **Controls**: `GameControls.vue` transcript panel restyled for legibility.
> - **Core Logic**:
>   - **Answer Validation**: `useGameQuestions.js` now returns `[isCorrect, foundWord]`; `useGameCore.js` consumes and exposes `isAnswerPlaying`, `isCorrect`, `firstMatchingAnswer`.
>   - **Car Counting**: `useCarCounting.js` supports digit/word answers and returns format-aligned answers; `CarCounting.vue` uses config and answer-only captions.
> - **Configs & Assets**:
>   - **New Game Config**: `gameConfigs.js` adds `carCounting`; updates several `iconSrc` to new PNGs.
>   - **GameZone Catalog**: `GameDB.js` icon paths updated.
>   - Removes unused SVGs (`impact/carousel/*Chevron.svg`, `testimonials/yellowStar2.svg`).
> - **Troubleshooting Overhaul**:
>   - **Content & RWD**: `troubleshooting.json` restructured (HTML text, step images via `prefix/suffix`, added `iconPath`); `Troubleshooting.vue` implements device-aware images, improved image viewer, and new styling.
> - **Site UI/Styling**:
>   - **Buttons & Utilities**: Consolidates to `page-button`/`page-button-flex`; replaces legacy classes across pages; minor accessibility tweaks (`aria-hidden`, decorative SVGs hidden).
> - **Pages & Content**:
>   - **About/Volunteers**: New `VolunteerStaff` component replacing `VonlunteerStaff`; content/image updates in `Staff.js`; `CarlsJourney.vue` layout refresh.
>   - **Press**: Adds new White House press item; header/button styling updates.
> - **Game Data**:
>   - **Odd One Out DB**: Rewrites many prompts to use varied outliers (replacing repeated "chair").
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 00295cf8cedd1eec9c5bc0e3c5aed3c93f1635c9. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

jschhie committed:
refactor: Replace JS device checks w/ Tailwind RWD

- Replace 'checkDeviceType.js' import w/ native classes for performance
- Adjust RWD margins & video scaling
- Remove unused <ProjectsInAction />

jschhie committed:
refactor: DRY styles & fix RWD icons for <OurReach />

- Apply shared classes & add <PageDecorations /> for RWD icons
- Add aria-hidden & remove alt text for screen reader access.
- Remove unused image imports

jschhie committed:
fix: RWD in <CollaboratingSchools/>

- Fix text & button overlap/cutoff during resizing
- Fix RWD & adjust/extract styles for <Map/> & <CollaboratingSchools/>

jschhie committed:
accessibility: Hide decorations in <SchoolsList/> & <TeachersList/>

jschhie committed:
feat: Add <PasswordChecklist/> UI to ResetPassword

- Replace red error texts w/ <PasswordChecklist/> for UI consistency
- (non-functional, UI only)

jschhie committed:
feat: + Live Reset PW feedback & checklist logic (WIP: PII check)

- Add Logic for live PW match & strength feedback via watch()
- Use usePasswordCriteria.js & usePasswordFeedback.js composables
- Add validatePasswords() & update resetConfirm() in ResetPassword

PrajwalaMugajjiShambulingappa committed:
Added World's First Audio Console tab

jschhie committed:
refactor: Prep RWD for login/logout button in UserAuthHeader.vue

- Fix RWD layout for user auth button (to test <Header/>)
- Adjust inline comments & code format for clarity in <Header/>

jschhie committed:
feat: GET API for PII checks in Reset PW

- Add new API to validate token & fetch name/email for PII checks
- Update onMounted(): Add 'reset_password_get_user_basics' API call
- Update robots.txt & vercel.json for new API

jschhie committed:
feat: Disable Reset PW for invalid/expired tokens

- Update onMounted() & resetConfirm() to disable form & hide submit btn
- Add 'isDisabled' prop to <PasswordToggle/> to adjust hover effects
- Remove disabled inputs & button from tab order/live feedback for SRs
- Prevent extra password validation for invalid/expired tokens

jschhie committed:
fix: WCAG color contrast for auth form feedback UI

- Adjust bg colors for auth form error, password, & logout feedback UI
- Remove unused resetErrors() in Reset Password

jschhie committed:
fix: lg/xl RWD for logout btn in <UserAuthHeader/>

- Adjust RWD for lg (column) & xl (row) screens to fit text/profile
- Align account (login/logout) btn RWD for <Header /> & <UserAuthHeader />

jschhie committed:
feat: + <LoadingOverlay/> component in auth forms

- Improve UI, RWD, & SR access. in Log In, Sign Up, & Forgot/Reset PW forms

jschhie committed:
feat: Add game question captions in <GameHeader/>

- Add props/logic to <GameHeader/> to display accessible Qs & current index
- Use data from 'useGameCore.js' for dynamic & async updates

jschhie committed:
feat: Captions for Definition Detective game Qs

jschhie committed:
fix: RWD for question captions & style UI for <GameHeader/>

jschhie committed:
feat: Captions for non-MCQ Language games

- Update <GameHeader/> props passed in non-multiple choice Language games
- Games affected:
    - Part of Speech, Color Game, & Syllable Sorting

jschhie committed:
feat: Auto-format multi-choice Qs in <GameHeader/>

- Add 'multipleChoiceGames' array for MCQ games (ex: Vocab Vortex)

jschhie committed:
feat: Captions for Vocab Vortex w/ MCQ formatting

jschhie committed:
feat: Handle trailing 'or' in MCQs (<GameHeader/>)

- Update splitMCQs() to format/trim 'or' before last MC option (edge case)
- Adjust RWD flex for MCQs

jschhie committed:
feat: Captions for Polar Pairing w/ MCQ formatting

- NOTE: <GameHeader/> formats trailing 'or' in MC options (parent commit)

jschhie committed:
feat: Captions for Odd One Out w/ MCQ formatting

- Add game to 'multipleChoiceGames' & adjust RWD in <GameHeader/>

jschhie committed:
fix: Correct 'currentQuestion' prop in <GameHeader/>

- Fix prop type (string --> Object) & set default value (null)
- Add null check to display 'currentQuestion'

jschhie committed:
feat: Add question #s for Spelling Bee captions

- Add conditional logic to show current game Qn index & desc in <GameHeader/>

jschhie committed:
fix: Correct page URL in <WhatWeDo/> card

- Change /buzzle (page DNE) to /audio-console

jschhie committed:
style: Update <AudioConsole/> images; remove unused ones

jschhie committed:
fix: improve RWD & UI for <AudioConsole />

- Fix RWD spacing/resizing, update UI, & hide images for SR accessibility

jschhie committed:
feat: Add 'Buzzle' link & remove redundant 'Our Projects' in <Footer />

- Adjust RWD to toggle b/t shortened 'About Buzzle' (mobile-sm) & full text (medium+)

jschhie committed:
fix: mobile RWD & UI consistency for <AudioConsole />

- Fix narrow & text overlap for mobile card; align page typography

jschhie committed:
fix: RWD for <Header/> to fit new Buzzle link

- Fix RWD spacing & shorten text links for desktop/mobile <Header />

jschhie committed:
chore: Remove unused, accidentally generated files

- Remove two files accidentally generated via CLI output

jschhie committed:
fix: Missing arrow image paths on vercel preview

- Rename img path to remove whitespace for proper rendering
- Affected: <AudioConsole/>, <CollaboratingSchools/>, & <SocialMedia/>
- Minor refactor: Reuse shared Tailwind classes for UI consistency & RWD

jschhie committed:
refactor: mobile RWD for icons & text in <GameHeader />

- Improve RWD to reduce scrolling on mobile/small screens
- Minor updates: Hide icons from SR output & add RWD transitions

jschhie committed:
feat: Captions for Math games Qs

- Update <GameHeader/> props passed for each Math game
- Games affected:
    - Fruit Frenzy, Shape Shark, Animal Add.,
    - Sub. Safari, Multi. Madness,
    - Division Duel, & Monkey Madness

jschhie committed:
feat: Add question #s for Car Counting captions

jschhie committed:
feat: Captions for Science games Qs

- Update <GameHeader/> props passed in each Science game
- Affected: All 10 Science games

jschhie committed:
feat: Captions for Life Skills games Qs

- Update <GameHeader/> props passed in each Life Skill game
- Affected: All 14 Life Skill games

jschhie committed:
feat: Captions for Independence Skills games Qs

- Update <GameHeader/> props passed in each Indep. Skills game
- Affected: All 12 Indep. Skills games

jschhie committed:
feat: Logic/UI for multi-part games captions in <GameHeader/>

- Add splitMultiPartsQs(): Splits prompt from Q in <GameHeader/> for UI readability
- Games affected: Fruit Frenzy, Monkey Madness, & Shape Shark
- Minor refactor: Rename variable in splitMCQs() for clarity ('prompt' --> 'question')

DooHwanKim0419 committed:
feat: add special validation type to spellingBee

DooHwanKim0419 committed:
feat: extend validateAnswer to support spellingBee

DooHwanKim0419 committed:
refactor: refactor spellingBee to use gamecore logics

jschhie committed:
feat: Sidewide smooth scroll to page sections on button click

- Wrap html tag w/ 'scroll-smooth' class & re-run Prettier in index.html
- Affected pages: Audio Console & Game Toolkits

jschhie committed:
refactor: RWD & DRY classes for <ResearchTech/> in Impact page

- Replace icons & remove unused ones in <ResearchTech/>
- Re-use shared Tailwind classes

jschhie committed:
fix: Missing <ResearchListCard/> URLs & RWD

- Add UI logic for missing URLs & adjust RWD
- Rename <ResearchListCard/> vars for clarity & split title from subtitle for UI readability
- Remove <ResearchListCard/> icons from tab order to reduce redundant SR output

PrajwalaMugajjiShambulingappa committed:
Added games bought to life in accessibility studio

jschhie committed:
refactor: RWD & DRY UI for <HaveYouHeard/> in Impact page

- Fix RWD spacing & resizing; re-use DRY Tailwind classes
- Add UI logic to split <HaveYouHeardCard/> title from caption
  for UI readability
- Update card icons; remove unused ones

jschhie committed:
fix: Simplify RWD & DRY UI for <KatyYouthHacks/>

- Replace complex, WET RWD (hidden/block rules) for Carl img w/ DRY code
- Fix RWD spacing & resizing; re-use shared Tailwind classes
- Rename Carl img for clarity & fix missing path on Vercel preview

jschhie committed:
refactor: DRY UI & SR accessibility in <GamesShowcase/>

- Re-use Tailwind classes for DRY code & UI consistency
- Hide imgs & add dynamic 'aria-label' to buttons for SR output clarity
- Minor fixes: Game desc spelling & re-run Prettier

jschhie committed:
fix: RWD <GamesShowcase/> on mobile-medium screens

- Fix narrow cards/text; mobile-md: column & lg+: row layout

jschhie committed:
refactor: RWD & DRY UI for <AccoladesCard/>

- Adjust RWD spacing & resizing; re-use Tailwind classes

jschhie committed:
refactor: Improve <GamesShowcase/> RWD for game description

- Adjust RWD to prevent text cutoff & intuitive scroll in game desc
- Update game description for UI readability & style (via v-html)
- Add carousel arrow buttons RWD: up/down (mobile-md) & left/right (lg+)
- Minor: Update EchoQuest desc for clarity; add aria-labels to arrow btns

jschhie committed:
feat: Vertical & horizontal scroll for <GamesShowcase/> carousel

- Replace overflow-hidden to enable scroll on mobile & desktop (use overflow-x/-y-auto)
- Add scroll RWD: vertical (mobile-medium) & (large+ screens)

jschhie committed:
refactor: + DRY classes & formatted arrays in <GamesShowcase/>

- Add shared Tailwind RWD classes for carousel arrow buttons
- Replace long, inline classes w/ organized arrays (:class=[...]) for readability

jschhie committed:
refactor: DRY UI, format <Testimonials/>, & accessibility fixes

- Format testimonial text/emails for UI readability (via v-html)
- Add DRY style classes for page content & btns
- Minor SR accessibility: Hide imgs; add aria-labels to visual arrow btns

jschhie committed:
fix: RWD & add <Testimonials/> carousel scrolling

- Fix RWD for cards: column (mobile-md) & row (lg+)
- Add RWD for arrow btns: up/down (mobile-md) & left/right (lg+)
- Add vertical (mobile-md) & horizontal (lg+) scrolling to carousel
- Add 2xl RWD for <Testimonial/> buttons, text, & images
- Style: hover effects & RWD transitions

jschhie committed:
accessibility: Hide redundant linked <PressListCard/> images & remove them from tab order

- Hide images & remove their repeated links from SR output

jschhie committed:
refactor: RWD & DRY UI in <OurProjects/> carousel

- Fix RWD to avoid icon/image overlap in <ProjectsInAction/>
- Improve RWD spacing/resizing in <OurProjects/> & <Carousel/>
- Add DRY classes & update UI in <ProjectsInAction/> & <Carousel/>
- Accessibility: Hide decorative images & <Carousel/> from SR output

jschhie committed:
fix: Remove duplicate Science games from DB & Game Zone cards/menu

- Remove 8 duplicate Science games from GameDB.js
- Fix Science page to show 10 Game Zone cards/menu links
- NOTE: Duplicates were likely caused by a bad merge in main/preprod
- Chore: Remove extra newlines in GameDB.js

master-of-none committed:
chore: prettier fixes

jschhie committed:
fix: Blurry <ResearchListCard/> & Impact accessibility

- Replace previous images and remove them from repo

Minor accessibility & refactor:
- Hide <Card/> & <ResearchTechCard/> imgs,
- Add title to <SocialMedia/>, & fix <AccoladesCard/> header
- Refactor <SocialMedia/> inline arrow classes for readability

Crustaly committed:
Merge pull request #262 from Crustaly/fix/impact-page-RWD-research-cards-URLs

fix: RWD for Impact Page components & missing `<ResearchListCard/>` URLs

jschhie committed:
refactor: Update <GamesShowcase/> placement in Access. Studio

- Move <GamesShowcase/> above Problem/Opportunity section

jschhie committed:
fix: Correct AE Foundation text & logo

- Replace incorrect logo/text & remove from repo

Crustaly committed:
Merge pull request #263 from Crustaly/fix/AE-Foundation-logo-reorder-games-showcase-placement

hotfix: Correct AE Foundation text/logo & reorder `<GamesShowcase/>` placement

jschhie committed:
refactor: Update game icons for UI consistency

- Replace previous ones and remove them from repo
- Affected games: Language, Math, Science, Life Skills, & Indep. Skills

jschhie committed:
fix: Keyboard access for Our Projects <Carousel/>

- Fix UI logic for <Carousel/> buttons to be accessible via tab & Enter
- Replace inaccessible <img/> & complex event listeners:
  - Replace static values with reactive currentSlide, carousel, & maxLength
  - Add semantic <button @click=...> & goToNext/PrevSlide() functions
- Add image mappings (src & alt text) for dynamic <Carousel/> SR output
- Update <Carousel/> for better RWD, DRY code, & UI consistency

jschhie committed:
fix: Enable <Carousel/> previous button circularity

- Before: Previous button on first image did nothing
- After: Circular behavior; fast forwards to last image
- to align with Next button on last image (back to first)

jschhie committed:
fix: <Testimonials/> & <GamesShowcase/> legibility

- Add white BG to text in both carousels
- Center text in <Testimonials/> for UI balance
- Minor: + hover effect to <GamesShowcase/> to align UI w/ <Testimonials/>

jschhie committed:
chore: Re-run Prettier to fix formatting

jschhie committed:
chore: Remove unused images from repo

jschhie committed:
fix: Replace blurry game icons & remove previous ones from repo

jschhie committed:
chore: Fix typos in VolunteerStaff folder & file

- Rename directory & <VolunteerStaff/> for clarity

jschhie committed:
fix: RWD for <CarlsJourney/> & DRY UI

- Fix RWD layout for <CarlsJourney/> & add DRY Tailwind classes
- Accessibility: Hide decorative images & BG
- Update UI for style & consistency

jschhie committed:
fix: RWD for <VolunteerStaff/> & DRY UI

- Fix RWD layout for better spacing & resizing
- Affected: <Volunteers/> & <VolunteerStaff/>
- Add DRY Tailwind classes
- Minor: Update UI for consistency

jschhie committed:
fix: mobile & medium RWD for Impact page buttons

- Wrap buttons in flex & fix mobile/md rules
- Add DRY classes for buttons

Affected buttons:
- Get Affiliated, Check upcoming Events,
- Try out our games, & Watch more videos

jschhie committed:
fix: Incorrect <VolunteerStaff/> & Vercel img path

- Remove incorrect staff (DNE) from 'Staff.js' database
- Rename staff image to fix missing Vercel preview; remove outdated img
- Minor: Update staff img for UI consistency; remove prev img w/ purple bg

jschhie committed:
fix: transcription RWD & legibility in <GameControls/>

- Add transparent bg; update UI for better RWD & style
- Re-run Prettier

jschhie committed:
feat: Answer captions & progress bar in <GameHeader/>

Update useGameCore.js:
- Add reactive, timed 'isAnswerPlaying' & 'isCorrect' flags
- Add new flags to return values

Update <GameHeader/>:
- Pass matching props for dynamic answer captions
- Add UI logic for question progress bar
- Rename 'showQuestions' prop to 'showCaptions' for scope/clarity

Minor Refactor:
- Update RWD for new feats
- Extract classes for DRY code & readability; add inline docs

jschhie committed:
feat: Answer captions for Language Games

- Pass new props to <GameHeader/> for Fill-in-the-Blank Lang. Games

Affected:
- All Language Games (MCQ & Fill-in-the-Blank Qns)
- (Except Spelling Bee; handled separately)

jschhie committed:
fix: Extra 'image' SR output in <VolunteerStaff/>

- Add 'aria-label' to social <a> links
- Set empty 'alt' on <img> tags
- Remove redundant & unhelpful 'image' SR output

jschhie committed:
feat: Answer captions for Spelling Bee

- Add 'showAnswerOnly' prop in <GameHeader/> to hide Qns
- Flag affects: Spelling Bee & Car Counting games
- Pass matching props to <GameHeader/> for Spelling Bee

jschhie committed:
feat: Answer captions for Math games

- Pass matching props to <GameHeader/>
- Affected: All Math Games (Multi-part & Fill-in-the-Blank Qns)
- (Except Car Counting; handled separately)

jschhie committed:
feat: Answer/feedback captions for Car Counting

- Add reactive, timed 'isCorrect' & 'isAnswerPlaying' flags in CarCounting.vue
- Pass props to <GameHeader/>
- Add UI logic for Car Counting answer captions in <GameHeader/>

jschhie committed:
feat: Answer/feedback captions for Science games

- Pass props to <GameHeader/> for all 10 Science games

jschhie committed:
feat: Answer/feedback captions for Life Skills games

- Pass props to <GameHeader/> for all 14 Life Skills games

jschhie committed:
fix: Align <GameHeader/> captions for synonyms & #s

- Align answer captions w/ transcription for synonyms and # formats
- Add 'foundWord' return value to validateAnswer() for matching
- Implement 'firstCorrectAnswer' prop & UI logic in <GameHeader/>
- Affected: All Math games (except Car Counting) & Grocery Grab
- Minor: 2xl RWD for captions

jschhie committed:
fix: Answers & captions w/ digit-based numbers for Car Counting

- Add reactive 'isWordBasedFormat' flag & 'wordToDigitMap' in useCarCounting.js
- Fix validateCarAnswer() to check if transcript has digits
- Fix getCurrentAnswer() to align answer captions w/ transcript word/digit format
- Rename 'answerMap' to 'digitToWordMap' for clarity
- Minor: Refactor inline <GameHeader/> classes into constants

jschhie committed:
refactor: Add/Use gameConfigs.js for Car Counting

- Add missing game entry in gameConfigs.js
- Replace hardcoded values w/ gameConfigs in carCounting.vue

jschhie committed:
feat: Answer/feedback captions for Indep. Skills games

- Pass props to <GameHeader/> for all 12 Independence Skills games

jschhie committed:
fix: 40 duplicate Odd One Out 'chair' answers

- Replace 40 duplicate 'chair' answers with alternative choices
- Improve QnA variety for better gameplay/UX

jschhie committed:
refactor: Rename 'firstCorrectAnswer' to 'firstMatchingAnswer'

- Improve variable clarity; no logical or UI changes

NOTES:
- Code finds 1st matching answer to transcript, not 1st answer in array
- Variable prevents UI mismatch in answer captions for <GameHeader/>

jschhie committed:
fix: mobile & medium RWD for Home page buttons

- Wrap buttons in flex & fix mobile/md RWD rules
- Affected: Play our Games; Explore..., Discover..., See More/Less
- Minor refactor: DRY UI for PressList.vue

jschhie committed:
fix: mobile & medium RWD for Toolkits buttons

- Affected btns: View All; Download...; Partner...
- Hide <PageDecorations/> on mobile (sm+: block) for better spacing & RWD

jschhie committed:
fix: mobile & medium RWD for Access. Studio buttons

- Affected btns: Start Test & Book Test
- Hide <PageDecorations/> on mobile (sm+: block) for better spacing & RWD

jschhie committed:
fix: mobile & medium RWD for Buzzle page buttons

- Affected btns: Learn More
- Use DRY classes from 'style.css' in 'AudioConsole.vue'
- Remove redundant, unused classes from 'game-resources-style.css'
- Hide <PageDecorations/> on mobile (sm+: block) for better spacing & RWD
- Fix RWD for SVG wave

jschhie committed:
feat: Hover effects to Game Page buttons

- Improve UI consistency & UX; no logical changes

Affected btns:
- Play/Start Q, Go Back, Something Not Working,
- Record/Stop, Repeat Q

jschhie committed:
feat: Add logic for synonyms in Color Game answers

- Pass 'firstMatchingAnswer' to <GameHeader/> to align captions w/ transcript
- Update 'crazyColorsDB.json' with synonymous answers & fix grammar
- Add new Hard Q&As for better variety & gameplay UX

jschhie committed:
feat: + Logic for synonyms & new Q&A in Life Skills games

- Pass 'firstMatchingAnswer' to each <GameHeader/> for UI caption alignment
- Update Life Skills Q's (.json) w/ synonyms & new Easy-Hard Q&A
- Affected: All 14 Life Skills games

jschhie committed:
feat: + Logic for synonyms & new QnA for Indep Skills

- Pass 'firstMatchingAnswer' to each <GameHeader/> for UI caption alignment
- Update Independence Skills Q's (.json) w/ synonyms & new Easy-Hard Q&A
- Affected games: All 12 Indep Skills games

jschhie committed:
chore: Remove redundant Math answer choices

- Remove repeated answer choices & hyphenated #'s in across Math games (.json)
- Affected games: Shape Shark, Fruit Frenzy, & Sub. Safari
NOTES:
- validateAnswer(): Auto-converts answers to lowercase
- Kept answers w/ titlecase, hyphens, or whitespace

jschhie committed:
fix: Logic for game answers with whitespace

- Fix validateAnswer() to check if answer has whitespace/phrase (2+ words)
- Rename 'foundWord' to 'foundMatch' for variable clarity
- Add conditional logic for 'whitespaceAnswers'
- Affected games (w/ whitespace answers):
  - Color Game & Space Case
  - All Math games (except Car Counting)
  - Money Match; Kitchen Cues; Safety Sirens
  - Phone Friend; Social Sense; Asking for Assistance

PrajwalaMugajjiShambulingappa committed:
Updated Press List with White House feature

Crustaly committed:
Merge pull request #271 from Crustaly/presslist-updates

Updated Press List with White House feature

jschhie committed:
fix: + Logic for synonyms in Space Case answers

- Pass 'firstMatchingAnswer' prop to <GameHeader/>
- Fix spelling & add synonyms for answer choices

jschhie committed:
refactor: Import <GameHeader/> constants for readability

- Move long UI classes & game title constants into new 'GameHeaderConstants.js'
- Improve <GameHeader/> script readability

jschhie committed:
feat: Import new <GameProgressBar/> in <GameHeader/>

- Refactor long template into new reusable <GameProgressBar/> component

jschhie committed:
feat: Import new <QuestionCaptions/> in <GameHeader/>

- Refactor long template into new reusable <QuestionCaptions/> component

jschhie committed:
feat: Import new <AnswerCaptions/> in <GameHeader/>

- Refactor long template into new reusable <AnswerCaptions/> component
- Minor: Update inline docs for GameHeaderConstants.js to reflect scope

jschhie committed:
refactor: Align props syntax in Game Header components w/ codebase

- Add 'const props = ...' to defineProps() for consistency
- Modified:
    - <GameHeader/>, <GameProgressBar/>,
    - <QuestionCaptions/>, & <AnswerCaptions/>

jschhie committed:
refactor: Use computed property for <AnswerCaptions/> text

- Replace inline, complex ternary exps. w/ computed 'captionText' for clarity
- Minor: Re-run prettier

jschhie committed:
feat: + Logic for representative <AnswerCaptions/>

- Add new 'hasRepAnswer' flag to questionsDb (ie, moneyMatchDb.json)
- Add logic to <AnswerCaptions/> to display 1st answer choice based on flag
- Improve UX by displaying:
    - Full names for proper nouns (ex: presidents, planets, AI tools)
- Affected games:
    - Space Case
    - Money Match
    - Phone Friend

DooHwanKim0419 committed:
fix: make sure that the first question is not skipped in mobile

PrajwalaMugajjiShambulingappa committed:
Changed Affiliated Organizations to Associated Organizations

Crustaly committed:
Merge pull request #273 from Crustaly/minor-UI-changes

Changed Affiliated Organizations to Associated Organizations

jschhie committed:
fix: RWD & DRY UI for Troubleshooting Page

- Fix RWD for page text & images during resizing & mobile/small screens
- Refactor repeated styles into DRY Tailwind classes
- Align captions (steps 1-5) w/ images & add inline classes in .json
- Add 'v-html' for better UI readability & hover effects
- Fix outdated email & add link
- Add hover effects & 'title' to clickable images for UI clarity

jschhie committed:
fix: RWD mobile/full images in Troubleshooting Page

- Add RWD logic to switch between new mobile & full images
- Rename full images for clarity & to support image RWD logic
- Add 'prefix' & 'suffix' in .json for RWD image paths
- Minor:
    - Add 'subtype' flag in .json for dynamic text styling
    - Update UI style for new tabs via openImageInNewTab()
    - Add clearer, updated images
    - Fix page text RWD

jschhie committed:
fix: Add dynamic icon paths to Troubleshooting Page

- Update .json to replace hardcoded image url with dynamic 'iconPath'
- Add decorative SVG wave; adjust RWD spacing

jschhie committed:
fix: mobile/small RWD Troubleshooting images

- Add new cropped images for better scaling on mobile/small screens
- Add medium RWD rules for images

jschhie committed:
refactor: DRY RWD logic for Troubleshooting images

- Refactor repeated, hardcoded RWD templates into DRY, dynamic logic
- Import/use 'useDeviceDetection' composable
- Add/use new 'getImagePath(prefix, suffix)' method for dynamic RWD image paths

jschhie committed:
fix: RWD image & favicon to new Troubleshooting tab

- Fix image RWD for better scaling on new tab
- Add missing favicon for UI consistency
- Minor: Refactor long, inline classes into JS constants

jschhie committed:
fix: Answer validation & captions for proper nouns

- Fix validateAnswer() to return capitalized answers via 'firstMatchingAnswer' (vs lowercased)
- Fixes <AnswerCaptions/> for proper nouns by using this new return value
- Remove now-redundant hardcoded 'hasRepAnswer' flags from 'Space Case' Q&A .json

jschhie committed:
fix: Language games 1st Qn timing on mobile & tablet

- Affected: All 8 Language games (.vue files)
- Add 'Start Questions' button on desktop to align current Qn index & UX
- Remove device-based logic from <StartQuestionsButton/> & <GameControls/>

'useGameCore.js' Updates:
- Remove moveToNextQuestion() call from startFirstQuestion()
- Remove device-based logic from 'numOfAudiosPlayed' (computed property)
- Remove desktop-only auto-play logic from 'playButton' watch()er
- Return 'currentQuestionIndex' & import it in game .vue files

<GameHeader/> Updates:
- Update 'showCaptions' conditionals
- Replace 'numOfAudiosPlayed' prop with 'currentQuestionIndex'

jschhie committed:
chore: Resolve preprod to main merge conflicts

- Merged latest 'main' into this branch, resolving conflicts
- Accepted preprod's latest changes in SpellingBee.vue, CarCounting.vue, & useCarCounting.js
- Added missing Game Toolkits PDF files from main

master-of-none committed:
Merge pull request #278 from Crustaly/chore/resolve-preprod-main-merge-conflicts

chore: Resolve `preprod` to `main` merge conflicts

jschhie committed:
fix: Math games 1st Qn timing on mobile & tablet

- See parent commit for details
- Affected: All 8 Math games, except Car Counting (handled separately)

jschhie committed:
fix: Car Counting 1st Qn timing on mobile & tablet

- Fix 'showCaptions' timing logic
- Add/use 'hasStartedFirstQuestion' reactive flag in startFirstQuestion()
- Add computed 'numOfAudiosPlayed' logic
- Add Start Questions button to desktop to align current Qn index and UX
- Remove auto-play 1st Qn on desktop in 'playButton' watch()er

jschhie committed:
fix: Science games 1st Qn timing on mobile & tablet

- Affected: All 10 Science games

jschhie committed:
fix: Life Skills games 1st Qn timing on mobile & tablet

- Affected: All 14 Life Skills games

jschhie committed:
fix: Indep Skills games 1st Qn timing on mobile & tablet

- Affected: All 12 Independence Skills games

jschhie committed:
fix: Def Detective synonym support for captions

- Pass 'firstMatchingAnswer' to <GameHeader/> to align captions w/ transcript
- Minor: Cleanup misleading game answers/homophones
    - Ex: 'ball' vs 'bawl'; 'teacher' vs 'teach her'

jschhie committed:
fix: Synonym logic & add digits in Syllable Sorting answers

- Add digit-based numbers to syllableGameDB.json
- Pass 'firstMatchingAnswer' prop to <GameHeader/>

jschhie committed:
fix: Captions/background z-index (UI stacking)

- Replace invalid '-z-1' with correct, non-native value '-z-[1]' (transparentBgClasses)
- Add 'z-0' to relative parent containers for proper bg/text UI stacking
- Affected components: <AnswerCaptions/> & <QuestionCaptions/>

DooHwanKim0419 commented:
@master-of-none I'll review this tomorrow. Thanks.

jschhie commented:
Hi @master-of-none @DooHwanKim0419 & @Crustaly, 

Could we please delay this merge for a bit until PR #280 is merged first? 

PR #280 addresses validation fixes (broken whitespace answers & improved synonym support). 

It would prevent us from reviewing known bugs in this PR (#277). 

**Thanks so much!**

DooHwanKim0419 committed:
refactor: remove unused imports from games

master-of-none commented:
> Hi @master-of-none @DooHwanKim0419 & @Crustaly,
> 
> Could we please delay this merge for a bit until PR #280 is merged first?
> 
> PR #280 addresses validation fixes (broken whitespace answers & improved synonym support).
> 
> It would prevent us from reviewing known bugs in this PR (#277).
> 
> **Thanks so much!**

Yes we can delay until the bugs are fixed correctly.

jschhie committed:
fix: Stats/Typos (Hero, Press List, Access. Studio, & README)

- Affected: <Hero/>, <PressList/>, <AccessibilityStudio/>
- Update stats & format text for UI style/readability in README.md

jschhie reviewed (approved):
# PR #277 Review (`preprod` → `main`)

**Approved!** 

---

## Testing Summary
- Games remain functional
- Sampled 1+ games per category
- Stats consistency/Typos are fixed on affected pages & `README.md`
- Checked RWD (`1680px` down to `350px`) for: 
  - Updated "Volunteers" & "Troubleshooting" Pages
  - Stylized game Q&A captions UI (MCQ, Fill-in-the-Blank, Prompt/Question) 
  
---

## Testing Details

> **Test Cases 2-5:** Applicable Games & Expected Answers

- [x] 1. **Gameplay Flow (mobile, tablet, desktop)**: 
  - [x] <kbd>Play</kbd> > Intro audio > <kbd>Start Questions</kbd> > Question # 1 and/or Progress Bar
  - [x] Exactly 5 Q's played per game on all views/during resizing
- [x] 2. **Whitespace answers:**
  - [x] **"Color Game":** Test inputs like "black and white" or "white and black"
  - [x] **"Shape Shark":** Test "right angled triangle", "regular hexagon"

- [x] 3. **Proper Noun UI Capitalization:** 
  - [x] **"Space Case":** UI renders "Milky Way"
  - [x] **"Money Match":** UI renders "George Washington"

- [x] 4. **"Representative" Answers:** (Flexible Matching / Stylized Spelling)
  - [x] **"Money Match":** Inputting any president's first/last/full name should be marked correct but UI must render full name
  - [x] **"Phone Friend":** Inputs like "voice over" & "talk back" should be marked correct but UI must render stylized answers ("VoiceOver" / "TalkBack")

- [x] 5. **Spelling Bee & Car Counting:** Captions for Q's should NOT appear for these special games. Only progress bar & answer feedback UI should be visible.

jschhie commented on src/components/Game/QuestionCaptions.vue:
```diff
@@ -0,0 +1,141 @@
+  const question = questionParts[1];
+
+  return { prompt, question };
+};
```
Will address this in a later PR/release (noted in Issue #282)

jschhie commented:
Hi @Crustaly @master-of-none & @DooHwanKim0419,  

PR #277 now has the required approvals 👍

But due to conflicts, the "Rebase & Merge" option is currently unavailable. 

I'm not sure if we should do a regular "Merge" just to update `main` faster. 

---

@master-of-none 
Could you please look into this? It looks like it involves a complex `git rebase`. 

Thanks!

master-of-none commented:
> Hi @Crustaly @master-of-none & @DooHwanKim0419,
> 
> PR #277 now has the required approvals 👍
> 
> But due to conflicts, the "Rebase & Merge" option is currently unavailable.
> 
> I'm not sure if we should do a regular "Merge" just to update `main` faster. 
> 
> @master-of-none Could you please look into this? It looks like it involves a complex `git rebase`. 
> 
> Thanks!

Rebasing this would require checking many commits since there are like 135 commits, it is very hard to check manually and do a rebease. 
A better option currently would be to create a merge commit and merge it.
